### DISCUSSION TITLE
Is there a way to execute the block script if is showed?

### DISCUSSION THREAD

**Discussion by hylink-carlos-nieto** [cearls/timberland#23] [Q&A]
Hi @cearls I'm going to use this theme in a new project and have a question about the custom script used in each block. Is there a way that the script only will be added in the app.js if the block is in use? I hope you understand this question.

Regards, Carlos.

**Comment by cearls** [comment]
Hi! No, app.js will import the js for all blocks. In my experience, it's normally pretty minimal when using alpine unless you're pulling in third-party libraries. It would be nice to be able to break the js up into chunks and provide each page exactly what it needs, but I'm not sure how I'd make the connection between Webpack and what WP blocks are included on a page.

Also, if you're starting a new project, I'd use the `block-json` branch. It's just about ready for a release.

[https://github.com/cearls/timberland/tree/block-json](https://github.com/cearls/timberland/tree/block-json)
